### PR TITLE
[WIP] - Upcoming v1

### PR DESCRIPTION
**Pull request by axe312ger**
See #30 

This is **not** ready yet, there is still some stuff to do!

Currently includes:
* Updated dependencies (except svgo, needs async logic first)
* Integrate eslint && prettier
* Add unit and e2e tests via [jest](https://facebook.github.io/jest/) and [nixt](https://github.com/vesln/nixt)
* Replace unmaintained argv with yargs. Allows more complex cli argument parsing. Especially helpful for later on.
* Refactors SVG optimization

More to come...

#### Introduced breaking changes:
* The input file for the CLI now has to be passed via the `-i` argument. `sqip foo.jpg` will become `sqip -i foo.jpg`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+language: node_js
+node_js:
+  - "6"
+  - "8"
+cache:
+  directories:
+    - /home/travis/build/technopagan/sqip/vendor
+before_install:
+  - npm install -g npm
```
Comment (DanielRuf) [inline .travis.yml]: This is not needed as npm is provided with nvm ;-)

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+language: node_js
+node_js:
+  - "6"
+  - "8"
+cache:
+  directories:
+    - /home/travis/build/technopagan/sqip/vendor
```
Comment (DanielRuf) [inline .travis.yml]: "vendor"? Just know this from PHP. Is this really needed?

File: .travis.yml (hunk 1 above)
Comment (axe312ger) [reply .travis.yml]: Absolutely true. Thats a case of copy pasta 🙈

File: .travis.yml (hunk 2 above)
Comment (axe312ger) [reply .travis.yml]: U are probably correct here, too. I will remove it and we will see :)

**Comment by axe312ger** [comment]
Since we now have a `legacy` branch, I gonna merge this to master to reflect the current progress :)

**Pull request merged**